### ISSUE TITLE
New getenvoy version breaks old install methods using CLI + unpublishes older versions that were locked prior with no option to stay on older version

### ISSUE THREAD

**Issue by nheaps-chwy** [tetratelabs/getenvoy#261]
**Describe the bug**
A new version of getenvoy was released today that changes the command `fetch` to `use` and provides a new manifest that doesn't have older versions of envoy that were previously version locked. No option is provided to remain on the older version using the older manifest that references version 1.12.0.

Yes I recognize that the version is very old but upgrading that is currently out of scope due to the additional configuration changes and testing required (though it is in our long term roadmap)

**To Reproduce**
1. Run the old way of getting envoy `curl -L https://getenvoy.io/install.sh | bash -s -- -b /usr/local/bin`

**Expected behavior**
Installation steps for installing old versions still work, or at least the old version still exists as there's been no noticable deprecation notice (I have kept coming back to check every time bintray had an issue)

**Environment (please complete the relevant information):**
 - OS: AWS linux docker image

**Additional context**
We've been unblocked by converting to the RPM/Yum installation method which has 1.12.0 available. This is mainly to provide context to others that are also experiencing the issue

**Comment by codefromthecrypt** [comment]
Hi, @nheaps-chwy thanks for reporting! We did have to change things including the format order to accommodate arm64, etc and fix some problems. While the biggest change is behind us, but I appreciate this came as a surprise to you. When we do version 1.0 (hoping very soon) all syntax and formats will be portable on each release.

## On quickstart script
https://raw.githubusercontent.com/tetratelabs/getenvoy/master/site/install.sh is a modified copy "godownloader" script that gets a download from the current release, right now https://github.com/tetratelabs/getenvoy/releases/tag/v0.3.0

While this does accept a tag argument (like v0.3.0), it doesn't work with old releases. It isn't really a package manager and the [godownloader project](https://github.com/goreleaser/godownloader) is abandoned. I had to change the script last night just to keep it working. I might also rewrite the downloader script for reasons like this, but not for the <0.3 versions.

To get an old release the same way as that script means finding the tag you want here and clicking "Assets". https://github.com/tetratelabs/getenvoy/releases or otherwise doing that in curl. The old versions are there.

The only stable package we have are brew packages. If you would use a RPM/YUM package of getenvoy we can add that. I'm surprised no one has asked, yet, so please do! We are trying to make changes based on end users, so it would help to raise that.

## On binary URLs
Hosting binaries permanently has been a problem in the past as you noticed. We aren't currently doing permalinks.

For your information, we decided to only host non-EOL versions in the main json file per here https://github.com/envoyproxy/envoy/blob/main/RELEASES.md#release-schedule We didn't convert EOL versions into the new format to stop encouraging them, and also relay that these are not permanent URLs.

In your case, that edged out 1.12, but it wasn't completely unexpected someone might ask for an EOL version. I raised https://github.com/tetratelabs/getenvoy/pull/262 to add back 1.12 and 1.13 last patches. 

FYI canonical version list is here https://raw.githubusercontent.com/tetratelabs/getenvoy/master/site/envoy-versions.json
but directly referenced via redirect in the default of `ENVOY_VERSIONS_URL=https://getenvoy.io/envoy-versions.json`

This URL can be overridden and we don't expect to make any breaking changes to [the format](https://raw.githubusercontent.com/tetratelabs/getenvoy/master/site/envoy_versions-schema.json) now.

For example, it should be much easier to host your own json now that the URL parameter is no longer an internal detail, and the format is published. That said, you shouldn't need to.

## In closing
Excellent issue! thanks for all the details it helps steer!

**Comment by codefromthecrypt** [comment]
TODO: update website (sadly in a different repo) explaining what to do when the install script doesn't install the version you want.

**Comment by codefromthecrypt** [comment]
https://github.com/tetratelabs/getenvoy/pull/264 describes better where versions come from and how to override

**Comment by codefromthecrypt** [comment]
added this issue to ask envoy to retain the binaries they already build https://github.com/envoyproxy/envoy/issues/16830 This hopefully can bring long-term and permanent URLs for tarballs

**Comment by codefromthecrypt** [comment]
I'll close this out with this which has all available versions now https://github.com/tetratelabs/getenvoy/pull/282
